awx/ui_next/src/components/Lookup/MultiCredentialsLookup.jsx

```diff
@@ -1,146 +1,167 @@
-          }
-        });
-      });
-      this.setState({ credentialTypes });
-    } catch (err) {
-      onError(err);
-    }
-  }
+async function loadCredentialTypes() {
+  const { data } = await CredentialTypesAPI.read();
+  const acceptableTypes = ['machine', 'cloud', 'net', 'ssh', 'vault'];
+  return data.results.filter(type => acceptableTypes.includes(type.kind));
+}
 
-  async loadCredentials(params) {
-    const { selectedCredentialType } = this.state;
-    params.credential_type = selectedCredentialType.id || 1;
-    return CredentialsAPI.read(params);
-  }
+async function loadCredentials(params, selectedCredentialTypeId) {
+  params.credential_type = selectedCredentialTypeId || 1;
+  const { data } = await CredentialsAPI.read(params);
+  return data;
+}
 
-  toggleCredentialSelection(newCredential) {
-    const { onChange, credentials: credentialsToUpdate } = this.props;
+function MultiCredentialsLookup(props) {
+  const { history, tooltip, value, onChange, onError, i18n } = props;
+  const [credentialTypes, setCredentialTypes] = useState([]);
+  const [selectedType, setSelectedType] = useState(null);
+  const [credentials, setCredentials] = useState([]);
+  const [credentialsCount, setCredentialsCount] = useState(0);
 
-    let newCredentialsList;
-    const isSelectedCredentialInState =
-      credentialsToUpdate.filter(cred => cred.id === newCredential.id).length >
-      0;
+  useEffect(() => {
+    (async () => {
+      try {
+        const types = await loadCredentialTypes();
+        setCredentialTypes(types);
+        setSelectedType(
+          types.find(type => type.name === 'Machine') || types[0]
+        );
+      } catch (err) {
+        onError(err);
+      }
+    })();
+  }, []);
 
-    if (isSelectedCredentialInState) {
-      newCredentialsList = credentialsToUpdate.filter(
-        cred => cred.id !== newCredential.id
-      );
-    } else {
-      newCredentialsList = credentialsToUpdate.filter(
-        credential =>
-          credential.kind === 'vault' || credential.kind !== newCredential.kind
-      );
-      newCredentialsList = [...newCredentialsList, newCredential];
-    }
-    onChange(newCredentialsList);
-  }
+  useEffect(() => {
+    (async () => {
+      if (!selectedType) {
+        return;
+      }
+      try {
+        const params = parseQueryString(QS_CONFIG, history.location.search);
+        const { results, count } = await loadCredentials(
+          params,
+          selectedType.id
+        );
+        setCredentials(results);
+        setCredentialsCount(count);
+      } catch (err) {
+        onError(err);
+      }
+    })();
+  }, [selectedType]);
 
-  handleCredentialTypeSelect(value, type) {
-    const { credentialTypes } = this.state;
-    const selectedType = credentialTypes.filter(item => item.label === type);
-    this.setState({ selectedCredentialType: selectedType[0] });
-  }
+  const isMultiple = selectedType && selectedType.name === 'Vault';
```

Comment:
See https://github.com/ansible/awx/pull/5408/files#r353261983.  Might be better to check `selectedType.kind === 'vault'`